### ISSUE TITLE
Cannot convert NULL to SEQUENCE

### ISSUE THREAD

**Issue by thinkong** [rethinkdb/rethinkdb-go#210]
last time I asked about a similar error... but that was string to sequence.

I am passing a SomeStruct0.. which contains.. a slice like `[]*SomeStruct1`
and then SomeStruct1 contains `map[int]*SomeStruct2`

when i try to insert this into the db it keeps saying 
Cannot convert NULL to sequence.. if i add `gorethink:"-"` to that specific slice it works... am i using this wrong again?

**Comment by thinkong** [comment]
here is a simple code that i might be doing wrong

```
type ccc struct {
    A int
}
type aaa struct {
    A1 map[int]*ccc
    A2 map[int]*ccc
}
type bbb struct {
    B1 []*aaa
}

// SaveMatchInfo saves match info
func Save() {
    b := bbb{}
    b.B1 = make([]*aaa, 2)
    for i := 0; i < 2; i++ {

        b.B1[i] = &aaa{}
        b.B1[i].A1 = make(map[int]*ccc)
        b.B1[i].A2 = make(map[int]*ccc)
        for j := 0; j < 10; j++ {
            b.B1[i].A1[j] = &ccc{}
            b.B1[i].A2[j] = &ccc{}
        }
    }
    query := r.Table("example").Insert(b)
    res, insErr := query.RunWrite(session)
    if insErr != nil {
        log.Println("insert failed", insErr.Error())
    }
    log.Printf("%#v\n", res)
}
```

**Comment by dancannon** [comment]
Hi,

Thanks for posting this issue, after looking at your code example I realised there are two issues here:
- Encoding fails silently, errors should at least be logged
- Maps with non-string keys are not supported. This is because JSON only supports objects with string keys. I will create a new ticket for converting basic types to strings for map keys.

To fix this issue I would recommend the ints to strings, I know this is not the best solution but its probably the fastest.

**Comment by thinkong** [comment]
@dancannon  oh.. ok.. Thanks for the fast help

are both issues part of gorethink problem? or is it some other problem?

**Comment by dancannon** [comment]
The first issue is a gorethink problem however I am not sure how to return the error to the user. I will look into improving this in the next major release.

The second issue is partially a gorethink issue, even the JS driver doesn't support non-string keys. However unlike the JS driver gorethink doesn't convert the keys to strings for you. The `encoding/json` package is the same for example http://play.golang.org/p/-9LSQuEBDe.

**Comment by dancannon** [comment]
This should now be fixed in 1.1.0

**Comment by thinkong** [comment]
so.. can I now use maps with integer indexes?
